setup.py

```diff
@@ -18,6 +18,9 @@
     logger.warning("README file not found or unreadable.")
     long_description = "See https://github.com/brunns/mbtest/"
 
+with open('requirements.txt') as f:
```

Comment:
Is there some way we can do without this? This is how I used to do it, but there is at least one downstream service which needs `setup.py` to be self contained, IIRC.

Reply:
Might be that we just have to live with having two copies of the requirements list - not really a biggie.

Reply:
Actually, if `tox.ini`, `setup.py` and `requirements.txt` can live independently of each other, it may simplify supporting both non-Dockerised and Dockerised workflows - the Docker containers would rely on the `requirements.txt` files, and tox would keep on doing its thing. Do you see any potential issues there?

Reply:
I'd like something in the build ensuring the two are in sync. If we have that, I can live with the duplication.

Reply:
Or, better still, can we build `requirements.txt` from `setup.py`, make it a generated artifact?

Reply:
One potential option: https://github.com/jazzband/pip-tools#example-usage-for-pip-compile

Reply:
Looks worth a try!

Reply:
[Travis](https://travis-ci.org/github/brunns/mbtest) is unhappy. I'm not sure why that's not showing up in the PR - I'll look into that.